integrations/GA/browser.js

```diff
@@ -1,122 +1,986 @@
+        this.track(rudderElement, { nonInteraction: 1 });
+      }
+    } else {
+      var dimensionsArray = {};
+      for (let val of this.dimensions) {
+        dimensionsArray[val.from] = val.to;
+      }
+      var metricsArray = {};
+      for (let val of this.metrics) {
+        metricsArray[val.from] = val.to;
+      }
+      var contentGroupingsArray = {};
+      for (let val of this.contentGroupings) {
+        contentGroupingsArray[val.from] = val.to;
+      }
+      logger.debug("in GoogleAnalyticsManager page");
+
+      var category = rudderElement.message.properties.category;
+      var eventProperties = rudderElement.message.properties;
+      var name =
+        rudderElement.message.properties.category +
+        " " +
+        rudderElement.message.name;
+      var campaign = rudderElement.message.context.campaign | {};
+      var pageview = {};
+      var pagePath = path(eventProperties, this.includeSearch);
+      var pageReferrer = rudderElement.message.properties.referrer || "";
+      var pageTitle;
+      if (
+        !rudderElement.message.properties.category &&
+        !rudderElement.message.name
+      )
+        pageTitle = eventProperties.title;
+      else if (!rudderElement.message.properties.category)
+        pageTitle = rudderElement.message.name;
+      else if (!rudderElement.message.name)
+        pageTitle = rudderElement.message.properties.category;
+      else pageTitle = name;
+
+      pageview.page = pagePath;
+      pageview.title = pageTitle;
+      pageview.location = eventProperties.url;
+
+      if (campaign) {
+        if (campaign.name) pageview.campaignName = campaign.name;
+        if (campaign.source) pageview.campaignSource = campaign.source;
+        if (campaign.medium) pageview.campaignMedium = campaign.medium;
+        if (campaign.content) pageview.campaignContent = campaign.content;
+        if (campaign.term) pageview.campaignKeyword = campaign.term;
+      }
+
+      var payload = {
+        page: pagePath,
+        title: pageTitle,
+      };
+
+      var resetCustomDimensions = {};
+      for (var i = 0; i < this.resetCustomDimensionsOnPage.length; i++) {
+        var property = this.resetCustomDimensionsOnPage[i];
+        if (dimensionsArray[property]) {
+          resetCustomDimensions[dimensionsArray[property]] = null;
+        }
+      }
+
+      ga("set", resetCustomDimensions);
+      pageview = extend(
+        pageview,
+        setCustomDimenionsAndMetrics(eventProperties, this.inputs)
+      );
+
+      if (pageReferrer !== document.referrer) payload.referrer = pageReferrer;
+
+      ga("set", payload);
 
-    if (location) {
-      ga("rudder_ga.set", "location", location);
+      if (this.pageCalled) delete pageview.location;
+
+      ga("send", "pageview", pageview);
+
+      if (category && this.trackCategorizedPages) {
+        this.track(rudderElement, { nonInteraction: 1 });
+      }
+
+      if (name && this.trackNamedPages) {
```

Comment:
the above also condition also calls: this.track(rudderElement, { nonInteraction: 1 });
do we need the condition then?

Reply:
we can incorporate both with or in single if. but it is required.